### ISSUE TITLE
[package] openssl/all: cmake target name is wrong

### ISSUE THREAD

**Issue by gocarlos** [conan-io/conan-center-index#952]
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **openssl/all**

### Steps to reproduce (Include if Applicable)

```shell
git clone https://github.com/gocarlos/cpp-jwt
cd cpp-jwt
mkdir build
cd build
conan install .. --build missing
# crash
cmake ..
cmake --build . -j
```

target name for openssl should be `OpenSSL::SSL` but it is `OpenSSL::OpenSSL`

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
......
CMake Error at tests/CMakeLists.txt:5 (add_executable):
  Target "test_jwt_object" links to target "OpenSSL::SSL" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
....
```

</details>

**Comment by gocarlos** [comment]
see https://cmake.org/cmake/help/v3.6/module/FindOpenSSL.html#imported-targets

**Comment by uilianries** [comment]
Hi @gocarlos !

Actually the find cmake file generated by Conan is not the exactly same provided by the projects. As you can read, [cmake_find_package](https://docs.conan.io/en/latest/reference/generators/cmake_find_package.html) uses `<PKG-NAME>::<PKG-NAME>`.

This request is not new, and it's under discussion. Please, take a look on this PR first: https://github.com/conan-io/conan-center-index/pull/486
The idea there was bringing exactly what you need, but only for OpenSSL.

However, it's a general problem, so there is an open discussion here: https://github.com/conan-io/conan-center-index/issues/419

Regards!

**Comment by gocarlos** [comment]
can this now be achieved after 1.25?

**Comment by uilianries** [comment]
@gocarlos Yes, totally!

**Comment by gocarlos** [comment]
humm, so in the [blog](https://blog.conan.io/2020/05/07/New-conan-release-1-25.html) @danimtb presented the solution for component based openssl. can we take this, or is there something missing?

**Comment by uilianries** [comment]
Not all parts. AS described on Components reference, it's not a real case: https://docs.conan.io/en/latest/reference/conanfile/attributes.html#cpp-info. For instance, we don't have components for zlib, also, those defines I think they don't exist.

**Comment by danimtb** [comment]
This feature is still not implemented in generators and it still an experimental feature until we got something working. You can still play with it in the cpp_info model but it will take some releases until we got something working in cmake

**Comment by uilianries** [comment]
Fixed by #1782
